reana/reana_dev/utils.py

```diff
@@ -335,3 +344,147 @@ def get_prefixed_component_name(component):
+        spec.loader.exec_module(module)
+        version = module.__version__
+
+    elif version_files.get(JAVASCRIPT_VERSION_FILE):
+        with open(version_files.get(JAVASCRIPT_VERSION_FILE)) as f:
+            package_json = json.loads(f.read())
+            version = package_json["version"]
+
+    return version
+
+
+def bump_semver2_version(current_version, part=None):
+    """Bump a semver2 version string.
+
+    :param current_version: current version to be bumped
+    :type current_version: str
+
+    :return: String representation of the next version
+    :rtype: string
+    """
+    if not semver.VersionInfo.isvalid(current_version):
+        click.echo(
+            f"Current version {current_version} is not a valid semver2 version. Please amend it"
+        )
+
+    parsed_current_version = semver.VersionInfo.parse(current_version)
+    next_version = ""
+    if parsed_current_version.build or part == "build":
+        next_version = parsed_current_version.bump_build()
+    elif parsed_current_version.prerelease or part == "prerelease":
+        next_version = parsed_current_version.next_version("prerelease")
+    elif parsed_current_version.patch or part == "patch":
+        next_version = parsed_current_version.next_version("patch")
+    elif parsed_current_version.minor or part == "minor":
+        next_version = parsed_current_version.next_version("minor")
+    elif parsed_current_version.major or part == "major":
+        next_version = parsed_current_version.next_version("major")
+
+    return str(next_version)
+
+
+def bump_pep440_version(current_version, part=None):
+    """Bump a PEP440 version string.
+
+    :param current_version: current version to be bumped
+    :param part: part of the PEP440 version to bump (one of: [major, minor, micro, pre]).
+    :type current_version: str
+    :type part: str
+
+    :return: String representation of the next version
+    :rtype: string
+    """
+    try:
+        version = Version(current_version)
+        next_version = ""
+        if version.pre or part == "pre":
+            next_version = Version(
+                f"{version.major}.{version.minor}.{version.micro}{version.pre[0]}{version.pre[1]+1}"
+            )
+        elif version.micro or part == "micro":
+            next_version = Version(f"{version.major}.{version.minor}.{version.micro+1}")
+        elif version.minor or part == "minor":
+            next_version = Version(f"{version.major}.{version.minor}+1.0")
+        elif version.micro or part == "micro":
+            next_version = Version(f"{version.major}+1.0.0")
+
+        return str(next_version)
+    except InvalidVersion as e:
+        click.echo(
+            f"Current {current_version} is not a valid PEP440 version. Please amend it"
+        )
+
+
+def bump_component_version(component, current_version):
+    """Bump to next component version."""
+
+    def _replace_version_string(file_=None, current_version=None, next_version=None):
```

Comment:
Remove and use the implementation from https://github.com/reanahub/reana/pull/371/files#diff-f1d005e86d9c3c61bd4155479765040aR361